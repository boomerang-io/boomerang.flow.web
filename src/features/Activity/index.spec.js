import React from "react";
import queryString from "query-string";
import { createMemoryHistory } from "history";
import { fireEvent } from "@testing-library/react";
import { WorkflowActivity } from "./index";

const mockfn = jest.fn();

jest.mock("carbon-components-react/es", () => ({
  __esModule: true,
  DatePicker: () => <div />,
  DatePickerInput: () => <div />
}));

const props = {
  activityActions: {
    fetch: () => new Promise(() => {}),
    reset: mockfn
  },
  teamsActions: {
    fetch: () => new Promise(() => {})
  },
  match: {
    params: "testid"
  },
  location: {},
  history:{},
  teamsState: {
    isFetching: false,
    status: "success",
    error: "",
    data: [
      {
        id: "1",
        name: "testing-team",
        workflows: [
          {
            id: "2",
            name: "testing-workflow"
          }
        ]
      }
    ]
  }
};

describe("WorkflowActivity --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowActivity", () => {
    const { baseElement } = rtlRouterRender(<WorkflowActivity {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});

describe("WorkflowActivity --- RTL", () => {
  const basicQuery = { order: "DESC", page: 0, size: 10, sort: "creationDate" };
  it("Select status tab correctly", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const { getAllByRole } = rtlRouterRender(<WorkflowActivity {...props} history={history} />);

    const statuses = getAllByRole("tab");
    fireEvent.click(statuses[1]);
    expect(history.location.search).toBe("?" + queryString.stringify({ statuses: "inProgress", ...basicQuery }));

    fireEvent.click(statuses[3]);
    expect(history.location.search).toBe("?" + queryString.stringify({ statuses: "failure", ...basicQuery }));

    fireEvent.click(statuses[0]);
    expect(history.location.search).toBe("?" + queryString.stringify({ statuses: undefined, ...basicQuery }));
  });

  it("Filter by team", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const { getByLabelText, getByText } = rtlRouterRender(<WorkflowActivity {...props} history={history} />);

    fireEvent.click(getByLabelText("Filter by team"));
    fireEvent.click(getByText("testing-team"));
    expect(history.location.search).toBe("?" + queryString.stringify({ teamIds: "1", ...basicQuery }));
  });

  it("Filter by workflow", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const { getByLabelText, getByText } = rtlRouterRender(<WorkflowActivity {...props} history={history} />);

    fireEvent.click(getByLabelText("Filter by workflow"));
    fireEvent.click(getByText("testing-workflow"));
    expect(history.location.search).toBe("?" + queryString.stringify({ workflowIds: "2", ...basicQuery }));
  });

  it("Filter by trigger", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });
    const { getByLabelText, getByText } = rtlRouterRender(<WorkflowActivity {...props} history={history} />);

    fireEvent.click(getByLabelText("Filter by trigger"));
    fireEvent.click(getByText("cron"));
    expect(history.location.search).toBe("?" + queryString.stringify({ triggers: "cron", ...basicQuery }));
  });
});
