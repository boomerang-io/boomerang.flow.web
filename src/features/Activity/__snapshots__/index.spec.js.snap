// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowActivity --- Snapshot Capturing Snapshot of WorkflowActivity 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="container"
    >
      <header
        class="container"
      >
        <div
          class="container"
        >
          <section
            class="headline"
          >
            <p
              class="subtitle"
            >
              This is all of the
            </p>
            <p
              class="title"
            >
              Activity
            </p>
          </section>
          <section
            class="content"
          >
            <div
              class="bx--skeleton__placeholder summarySkeleton"
            />
          </section>
        </div>
      </header>
      <main
        class="content"
      >
        <nav>
          <div
            class="bx--tabs tabs"
            role="navigation"
          >
            <div
              aria-label="listbox"
              class="bx--tabs-trigger"
              role="listbox"
              tabindex="0"
            >
              <a
                class="bx--tabs-trigger-text"
                href="#"
                tabindex="-1"
              >
                All
              </a>
              <svg
                aria-hidden="true"
                focusable="false"
                height="6"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 10 6"
                width="10"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 6L0 1 .7.3 5 4.6 9.3.3l.7.7z"
                />
                <title>
                  show menu options
                </title>
              </svg>
            </div>
            <ul
              class="bx--tabs__nav bx--tabs__nav--hidden"
              role="tablist"
            >
              <li
                class="bx--tabs__nav-item bx--tabs__nav-item--selected"
                role="presentation"
                tabindex="-1"
              >
                <a
                  aria-selected="true"
                  class="bx--tabs__nav-link"
                  href="#"
                  role="tab"
                  tabindex="0"
                >
                  All
                </a>
              </li>
              <li
                class="bx--tabs__nav-item"
                role="presentation"
                tabindex="-1"
              >
                <a
                  aria-selected="false"
                  class="bx--tabs__nav-link"
                  href="#"
                  role="tab"
                  tabindex="-1"
                >
                  In Progress
                </a>
              </li>
              <li
                class="bx--tabs__nav-item"
                role="presentation"
                tabindex="-1"
              >
                <a
                  aria-selected="false"
                  class="bx--tabs__nav-link"
                  href="#"
                  role="tab"
                  tabindex="-1"
                >
                  Succeeded
                </a>
              </li>
              <li
                class="bx--tabs__nav-item"
                role="presentation"
                tabindex="-1"
              >
                <a
                  aria-selected="false"
                  class="bx--tabs__nav-link"
                  href="#"
                  role="tab"
                  tabindex="-1"
                >
                  Failed
                </a>
              </li>
              <li
                class="bx--tabs__nav-item"
                role="presentation"
                tabindex="-1"
              >
                <a
                  aria-selected="false"
                  class="bx--tabs__nav-link"
                  href="#"
                  role="tab"
                  tabindex="-1"
                >
                  Invalid
                </a>
              </li>
            </ul>
          </div>
          <div
            aria-hidden="false"
          />
          <div
            aria-hidden="true"
            hidden=""
          />
          <div
            aria-hidden="true"
            hidden=""
          />
          <div
            aria-hidden="true"
            hidden=""
          />
          <div
            aria-hidden="true"
            hidden=""
          />
        </nav>
        <section
          class="filters"
        >
          <div
            class="dataFilters"
          >
            <div
              style="margin-right: 1.4rem; width: 14.125rem;"
            >
              <div
                class="bx--multi-select__wrapper bx--list-box__wrapper"
              >
                <label
                  class="bx--label"
                  for="activity-teams-select"
                >
                  Filter by team
                </label>
                <div
                  class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
                  role="listbox"
                  tabindex="-1"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Open menu"
                    class="bx--list-box__field"
                    data-toggle="true"
                    role="button"
                    tabindex="-1"
                    type="button"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="activity-teams-select__menu"
                      aria-expanded="false"
                      autocomplete="off"
                      class="bx--text-input bx--text-input--empty"
                      id="activity-teams-select"
                      placeholder="Choose team(s)"
                      role="combobox"
                      value=""
                    />
                    
                    <div
                      class="bx--list-box__menu-icon"
                    >
                      <svg
                        aria-label="Open menu"
                        focusable="false"
                        height="16"
                        name="chevron--down"
                        preserveAspectRatio="xMidYMid meet"
                        role="img"
                        style="will-change: transform;"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                        />
                        <title>
                          Open menu
                        </title>
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="margin-right: 1.4rem; width: 14.125rem;"
            >
              <div
                class="bx--multi-select__wrapper bx--list-box__wrapper"
              >
                <label
                  class="bx--label"
                  for="activity-workflows-select"
                >
                  Filter by workflow
                </label>
                <div
                  class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
                  role="listbox"
                  tabindex="-1"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Open menu"
                    class="bx--list-box__field"
                    data-toggle="true"
                    role="button"
                    tabindex="-1"
                    type="button"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="activity-workflows-select__menu"
                      aria-expanded="false"
                      autocomplete="off"
                      class="bx--text-input bx--text-input--empty"
                      id="activity-workflows-select"
                      placeholder="Choose Workflow(s)"
                      role="combobox"
                      value=""
                    />
                    
                    <div
                      class="bx--list-box__menu-icon"
                    >
                      <svg
                        aria-label="Open menu"
                        focusable="false"
                        height="16"
                        name="chevron--down"
                        preserveAspectRatio="xMidYMid meet"
                        role="img"
                        style="will-change: transform;"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                        />
                        <title>
                          Open menu
                        </title>
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="width: 14.125rem;"
            >
              <div
                class="bx--multi-select__wrapper bx--list-box__wrapper"
              >
                <label
                  class="bx--label"
                  for="activity-triggers-select"
                >
                  Filter by trigger
                </label>
                <div
                  class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
                  role="listbox"
                  tabindex="-1"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Open menu"
                    class="bx--list-box__field"
                    data-toggle="true"
                    role="button"
                    tabindex="-1"
                    type="button"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="activity-triggers-select__menu"
                      aria-expanded="false"
                      autocomplete="off"
                      class="bx--text-input bx--text-input--empty"
                      id="activity-triggers-select"
                      placeholder="Choose trigger type(s)"
                      role="combobox"
                      value=""
                    />
                    
                    <div
                      class="bx--list-box__menu-icon"
                    >
                      <svg
                        aria-label="Open menu"
                        focusable="false"
                        height="16"
                        name="chevron--down"
                        preserveAspectRatio="xMidYMid meet"
                        role="img"
                        style="will-change: transform;"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                        />
                        <title>
                          Open menu
                        </title>
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div />
        </section>
        <div
          style="margin-top: 1rem;"
        >
          <table
            class="bx--skeleton bx--data-table tableSkeleton"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Workflow
                </th>
                <th>
                  Trigger
                </th>
                <th>
                  Initiated By
                </th>
                <th>
                  Start Time
                </th>
                <th>
                  Duration
                </th>
                <th>
                  Status
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <span />
                </td>
                <td>
                  <span />
                </td>
                <td>
                  <span />
                </td>
                <td>
                  <span />
                </td>
                <td>
                  <span />
                </td>
                <td>
                  <span />
                </td>
                <td>
                  <span />
                </td>
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
            </tbody>
          </table>
        </div>
      </main>
    </div>
  </div>
</body>
`;
